test/unit-tests/range.test.js

```diff
@@ -0,0 +1,62 @@
+const chai = require('chai');
+const expect = chai.expect;
+var sinon = require('sinon');
+const chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+let { openBrowser, closeBrowser, goto, range, setConfig } = require('../../lib/taiko');
+let { createHtml, removeFile, openBrowserArgs, resetConfig } = require('./test-util');
+
+describe('Color picker test', () => {
+  let filePath;
+
+  before(async () => {
+    let innerHtml = `
+        <div>
+            <input type="range" id="range-1" name="range" 
+                min="0" max="100" value='2'>
+            <label for="volume">Volume</label>
+        </div>
+     `;
+    filePath = createHtml(innerHtml, 'Range');
+    await openBrowser(openBrowserArgs);
+    await goto(filePath);
+    setConfig({
+      waitForNavigation: false,
+      retryTimeout: 100,
+      retryInterval: 10,
+    });
+  });
+
+  after(async () => {
+    resetConfig();
+    await closeBrowser();
+    removeFile(filePath);
+  });
+
+  it('Set Range value with Integer', async () => {
+    await range({ id: 'range-1' }).select(10);
+    expect(await range({ id: 'range-1' }).value()).to.be.equal('10');
+  });
+
+  it('Set Range value with Float value', async () => {
+    await range({ id: 'range-1' }).select(10.81);
+    expect(await range({ id: 'range-1' }).value()).to.be.equal('11');
+  });
+
+  it('Should throw error when value is non integer', async () => {
+    await expect(range({ id: 'range-1' }).select('10')).to.be.rejectedWith(
+      'The range value should be Int or Float. Please pass a valid value.',
```

Comment:
Please convert this to an Int or Float and then try it before validating.
This is because we don't want to enforce type safety in a langauge that is not type safe and we can do our best to convert it.
In the sample above '10' appears to be a valid value.
If the value is something like 'xy' then let's show the generic error.

```suggestion
      'The value xy is not between the input's range of 0-100',
```